### ISSUE TITLE
[DocDB] Prevent crashing the master if trying to restore a deleted table

### ISSUE THREAD

**Issue by sanketkedia** [yugabyte/yugabyte-db#14734]
Jira Link: [DB-4075](https://yugabyte.atlassian.net/browse/DB-4075)
### Description

1. Create a database with two tables say T1 and T2
2. Take a snapshot of this database via `yb-admin create_database_snapshot`
3. Drop table T1
4. Restore snapshot taken in (2) via `yb-admin restore_snapshot` - restore gets stuck infinitely trying to restore T1 as evident in the logs
5. Failover the master leader/restart it - new leader crashes since it is not able to find metadata of T1

While this scenario is not supported, we should prevent master from crashing if `TableInfo` is not found for the table. In particular, we could mark Restore as FAILED and then provide clear guidance on what to do in the event of a failed restore which in this case would be to recreate T1 instead of bringing the cluster down.

**Comment by yamen-haddad** [comment]
This issue is a duplicate of this [issue 14762](https://github.com/yugabyte/yugabyte-db/issues/14762).
I landed a [fix](https://github.com/yugabyte/yugabyte-db/commit/aecc6eb09fad01941d0a2c6e01136f9a8bbd880d) in 2.19.
I am working on backporting it to 2.14, 2.16, 2.18
@sanketkedia I will mark the issue as done once the backport is done. What do you think?
